apps/backend/src/users/users.service.ts

```diff
@@ -41,4 +35,37 @@ export class UsersService {
 
     return users;
   }
+
+  async findOne(userId: number) {
+    const user = await this.usersRepository.findOneBy({ id: userId });
+
+    if (!user) {
+      throw new BadRequestException('User not found');
+    }
+
+    const currentUser = getCurrentUser();
+
+    const currentStatus = currentUser.status;
+    const targetStatus = user.status;
+    switch (currentStatus) {
+      case Status.RECRUITER:
+        if (targetStatus === Status.ADMIN) {
+          throw new BadRequestException('User not found');
+        }
+        break;
+      case Status.APPLICANT:
+        if (currentUser.userId !== user.userId) {
```

Comment:
🔧Going back to my question about whether documents should have either 1) an `_id` and `id` field OR 2) a `_id` and `userId` field, this line of code only works if documents have a `userId` field, but the code on line 40 only works if the documents have a `id` field. Maybe I'm missing something?